File: src/1732-Find-the-Highest-Altitude/find_the_highest_altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
  let highest = 0
  let altitude = 0

  for (let a of gain) {
    altitude += a
    if (altitude > highest) {
      highest = altitude
    }
  }

  return highest
};

const gain = [-5,1,5,0,-7]

console.log(solution.countMatches(gain))

File: src/1732-Find-the-Highest-Altitude/find_the_highest_altitude.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int largestAltitude(std::vector<int>& gain) {
        int highest = 0;
        int altitude = 0;

        for (int a : gain) {
            altitude += a;
            if (altitude > highest) {
                highest = altitude;
            }
        }

        return highest;
    }
};

int main() {
    std::vector<int> gain = {-5, 1, 5, 0, -7};
    
    Solution solution;
    
    std::cout << solution.largestAltitude(gain);

    return 0;
}